source/features/reload-failed-proxied-images.tsx

```diff
@@ -0,0 +1,83 @@
+import select from 'select-dom';
+import features from '../libs/features';
+
+// This feature tries to fix the problem with the broken badges on GitHub
+// You can read more on a related issue here https://github.com/badges/shields/issues/1568
+
+// Reload Constants
+const ReloadTryInterval = 1000;
+const ReloadMaxTries = 4;
+
+// Check if image is from the GitHub's proxy
+function isGithubProxiedImg(image: HTMLImageElement): boolean {
+	return (/^https:\/\/camo\.githubusercontent\.com\/.*/).test(image.src);
+}
+
+// Check if image has zero dimensions
+function isZeroSizeImg(image: HTMLImageElement): boolean {
+	return (image.naturalHeight === 0 && image.naturalWidth === 0);
+}
+
+// Test if image loads correctly, and reload it if there is an error
+function test(image: HTMLImageElement): void {
+	// Check if image is on page's DOM
+	if (!document.contains(image) || !isZeroSizeImg(image)) {
+		return;
+	}
+
+	// Clone image object to check if image loads with no problems
+	// We need to clone the element so that we can later replace it on the page
+	const tester = image.cloneNode(false) as HTMLImageElement;
+
+	// Handle successful loading
+	tester.addEventListener('load', () => {
+		// If image on page is not loaded
+		if (isZeroSizeImg(image) && image.parentNode) {
+			// Update image
+			// Updating the image in an other way may result on an addition request if it is not cached
+			image.parentNode.replaceChild(tester, image);
+		}
+	});
+
+	// Handle failure of loading the image
+	// (maybe, the original image source failed to provide the image to the proxy in the 4 seconds time-frame)
+	tester.addEventListener('error', () => {
+		let tries = parseInt(image.dataset.brokenProxiedImage || '0', 10);
+
+		// Limit max tries
+		if (tries > ReloadMaxTries) {
+			// Here we could also try to directly load the image from the `image.dataset.canonicalSrc`
+			// but this will expose the user to the server that hosts the image
+			return;
+		}
+
+		// Update tries
+		image.dataset.brokenProxiedImage = String(++tries);
+
+		// Try again later
+		setTimeout(() => {
+			test(image);
+		}, ReloadTryInterval);
+	});
+
+	// Start testing
+	tester.src = image.src;
+}
+
+async function init(): Promise<false | void> {
+	// Get each image on the page
+	select.all('img').forEach(image => {
+		// If it is a proxied image
+		if (isGithubProxiedImg(image)) {
+			test(image);
+		}
+	});
+}
```

Comment:
>  an additional request will not cause that much trouble assuming that the image is cached and ready to be served.

Let's retry once, then.

If you use https://github.com/fregante/image-promise this whole feature can just be:

```ts
async function handleErroredImage({target}): Promise<void> {
	await delay(5000);
	try {
		// A clone image retries downloading
		// `loadImage` awaits it
		// If successfully loaded, the failed image will be replaced.
		target.replaceWith(await loadImage(target.cloneNode()));
	} catch {}
}

function init(): void {
	delegate('img[src^="https://camo.githubusercontent.com"]', 'error', handleErroredImage);
}

features.add({
	id: __featureName__,
	description: 'Retries downloading images that failed downloading due to GitHub limited proxying.',
	screenshot: false,
	init
});
```

Reply:
Based on my tests/debugging `delegate` does not work in this case. I think GitHub gabs the errors for statistics and the event does not bubble up to the body element.

Reply:
It works if you pass a `true` as the last parameter of `delegate` to make it a passive listener.